src/redux/books/books.js

```diff
@@ -1,23 +1,79 @@
 // action types
+const FETCH_BOOKS = 'FETCH_BOOKS';
 const ADD_BOOK = 'ADD_BOOK';
 const REMOVE_BOOK = 'REMOVE_BOOK';
 
+const URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/9AXCsCxdH0qCaJZzZe1J/books';
+
 const initialState = [];
+const fetchedState = [];
+let newBook = {};
+let delBook = {};
 
 // action creators
-export const addBook = ({ id, author, title }) => ({
-  type: ADD_BOOK,
-  payload: {
+export const fetchBooks = () => async (dispatch) => {
+  const response = await fetch(URL)
+    .then((res) => res.json())
+    .then((data) => data);
+  let temp = {};
+  const fetchedBooks = Object.entries(response);
+  fetchedBooks.forEach((item) => {
+    temp = {
+      id: item[0],
+      author: item[1][0].author,
+      title: item[1][0].title,
+    };
+    fetchedState.push(temp);
+  });
+  dispatch({
+    type: FETCH_BOOKS,
+    payload: temp,
+  });
```

Comment:
- [ ] (Optional) It would be nice to make the action creator a function and then call it in this function. Making it a pure function makes it easier to test the function.